snippets/base/admin/adminmodels.py

```diff
@@ -223,6 +223,18 @@ def change_view(self, request, *args, **kwargs):
             request.POST = post_data
         return super().change_view(request, *args, **kwargs)
 
+    def get_readonly_fields(self, request, obj):
+        if not request.user.is_superuser:
+            return self.readonly_fields + ('for_qa',)
+        return self.readonly_fields
+
+    def get_queryset(self, request):
+        queryset = super().get_queryset(request)
+        if request.user.is_superuser:
+            return queryset
+        else:
+            return queryset.exclude(for_qa=True)
```

Comment:
nit: I would go for `filter(for_qa=False)` since it might be easier to read instead of using a `NOT` of sorts.